### ISSUE TITLE
Cannot open <Drive>\<module>.node

### ISSUE THREAD

**Issue by Vaspra** [smt116/node-native-ext-loader#8]
All this loader appears to do (with/out basepath) is make my program look for the .node file at the root of my drive.

**Comment by smt116** [comment]
Do you use `rewritePath`? Could you share the rule and version of webpack?

**Comment by Vaspra** [comment]
Sure, I am using `"webpack": "^4.23.1",`
My rule is:
```
            {
                test: /\.node$/,
                loader: 'native-ext-loader',
                options: {
                    basePath: ['app.asar', 'renderer'],
                },
            },
```

This results in:

```
Uncaught Error: Cannot open E:\app.asar\renderer\opencv4nodejs.node: Error: Invalid package E:\app.asar
```

**Comment by Vaspra** [comment]
I feel like I've tried every combination of loader and argument, every one seems to deal in absolute paths, so my built project never works if I rename my development filestructure. :/

**Comment by smt116** [comment]
When using a `basePath` it will always [load the path in the runtime relatively to the `__dirname`](https://github.com/smt116/node-native-ext-loader/blob/master/index.js#L46). For some reason, `path.resolve(__dirname, ...)` resolves to the root path. Could you prepare a minimal reproduction repository?

Do you set [`node.__dirname`](https://webpack.js.org/configuration/node/#node__dirname) manually or indirectly via another package?

**Comment by Vaspra** [comment]
I'm in the process of trying to figure something out but this will be an interesting consideration for your package:

https://github.com/webpack/webpack/issues/1599

**Comment by smt116** [comment]
I believe that you just need to set `node.__dirname` to `false`. This is how webpack works and I'm not sure how I could "consider" this to the package.

**Comment by Vaspra** [comment]
I only mean as a footnote. Some googling has found a great many people having similar looking pathing issues, and webpack seems to have a strange default of setting `node.__dirname = undefined` which results in `__dirname = '/'`, and I'm not sure why. Obviously this package relies on `node.__dirname` being otherwise.

**Comment by Vaspra** [comment]
I appear to be much closer to the correct result, but I am getting an error that it is looking in:
```C:\...\Local\Programs\amethyst\resources\app.asar\app\app.asar\renderer\opencv4nodejs.node```
I'm unsure why it is trying the extra `app\app.asar`, I have tried adding both `'../../'` and `'..', '..'` to the beginning of my basePath array, but they don't seem to have any effect. Any ideas would be really appreciated!

**Comment by smt116** [comment]
I don't think `../../` would work on Windows (shouldn't be `..\..\`?). It is hard to help without the code ;)

**Comment by Vaspra** [comment]
I've finally solved it. It was just a case of clearing out all the weird attempts at everything I've tried, all the little copying of files and extra bits I've been banging my head against the wall with, and using your loader with all default options _with_ the option `node.__dirname: false` in my webpack.

This has been painful but thank you for your package and assistance, I can finally rest. 🎉

Closing

Edit: I was building for `target: 'electron-main'`
